### PR TITLE
Kafka Receiver - Dispose More Gracefully

### PR DESCRIPTION
**Pull request by pjwilliams2**
## Description
1. Fixes memory access violations that occur when the cancellation token is disposed and still being accessed by the Kafka consumer. 
2. Removes tracking collection nullability since it is initialized during construction and always clean up collection
3. Pass cancellation token to add method
4. Handle own operation cancelled exception in tracking task

### CODE REVIEW

**Comment by pjwilliams2** [comment]
This also addresses #121.

File: RockLib.Messaging.Kafka/KafkaReceiver.cs [hunk 1]
```diff
@@ -220,34 +221,38 @@ protected override void Dispose(bool disposing)
...
+            //otherwise, memory access exceptions can occur when the consumer accesses a disposed  cancellation token
+            _completeDispose.Wait();
+
+            _consumerCancellation.Dispose();
+            _trackingCollection.Dispose();
+
             if (_consumer.IsValueCreated)
             {
                 _consumer.Value.Close();
-                _consumer.Value.Dispose();
```
Comment (NeilSchiefer) [inline RockLib.Messaging.Kafka/KafkaReceiver.cs:226]: Is there an issue with disposing the consumer?

Comment (pjwilliams2) [reply RockLib.Messaging.Kafka/KafkaReceiver.cs:226]: Probably not, but technically redundant: https://github.com/confluentinc/confluent-kafka-dotnet/blob/master/src/Confluent.Kafka/Consumer.cs#L558

Comment (pjwilliams2) [reply RockLib.Messaging.Kafka/KafkaReceiver.cs:226]: Docs say to prefer `Close` over `Dispose`: https://github.com/confluentinc/confluent-kafka-dotnet/blob/master/src/Confluent.Kafka/IConsumer.cs#L654

**Comment by NeilSchiefer** [comment]
We probably should put a similar test in like the SQS receiver so we can have some coverage of these changes.

**Review by NeilSchiefer — APPROVED** [review]

**Review by eslutz — APPROVED** [review]

**Review by kristycurrier — APPROVED** [review]

**Pull request merged**